Twispay WooCommerce Plugin/twispay/includes/scripts.php

```diff
@@ -302,15 +302,22 @@ public function is_available() {
              */
             function process_payment( $order_id ) {
                 /* Extract the order; */
-                $order = new WC_Order( $order_id );
+//                $order = new WC_Order( $order_id );
+
+//                $actual_link = (isset($_SERVER['HTTPS']) ? "https" : "http") . "://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]";
+                $actual_link = $_SERVER['HTTP_REFERER'];
```

Comment:
you should not relay on `$_SERVER['HTTP_REFERER']`. this page can be called from other urls plus not all browsers will send a referrer header.